dashboard/modules/state/state_head.py

```diff
@@ -33,13 +36,21 @@ def __init__(self, dashboard_head):
     def _options_from_req(self, req) -> ListApiOptions:
         """Obtain `ListApiOptions` from the aiohttp request."""
         limit = int(req.query.get("limit"))
-        timeout = int(req.query.get("timeout"))
+        # Only apply 80% of the timeout so that
+        # the API will reply before client times out if query to the source fails.
+        timeout = int(int(req.query.get("timeout")) * 0.8)
```

Comment:
we can do the timeout purely on client side though.

Reply:
why 80% here, seems a bit arbitrary. What if client doesn't have timeout?

Reply:
SGTM. I will add this as a private option